### ISSUE TITLE
Data loss when reading dataframe [bug]

### ISSUE THREAD

**Issue by Darel13712** [h2oai/datatable#3205]
If you read data with fread, some values are lost along the way. This does not happen with any file, and exact conditions are not clear, but this does happen.

- How to reproduce the bug?
The example file `'lost_col.csv'` contains 4 columns, the last one is a playcount, but if you try to read the file, that value is nowhere to be found:
```python
import datatable as dt
dt.fread('lost_col.csv')

   | 0142ae97f6a6cb78b68b10cf93e91558474b1dbb  7746d775-9550-4360-b8d5-c37bd448ce01  "weird al" yankovic
   |                                     void                                  void                 void
-- + ----------------------------------------  ------------------------------------  -------------------
[0 rows x 3 columns]
```

if you add another line, fourth column is now not lost
```python
dt.fread('not_lost.csv')
   | C0                                        C1                                    C2                      C3
   | str32                                     str32                                 str32                int32
-- + ----------------------------------------  ------------------------------------  -------------------  -----
 0 | 0142ae97f6a6cb78b68b10cf93e91558474b1dbb  309c62ba-7a22-4277-9f67-4a162526d18a  beck                   330
 1 | 0142ae97f6a6cb78b68b10cf93e91558474b1dbb  7746d775-9550-4360-b8d5-c37bd448ce01  "weird al" yankovic    327
[2 rows x 4 columns]
```

But if you download the whole [dataset](http://mtg.upf.edu/static/datasets/last.fm/lastfm-dataset-360K.tar.gz) and try to read `usersha1-artmbid-artname-plays.tsv`, it will crash on this line with `Too few fields, expected 4, but found only 3`. You can see the full error message here https://github.com/Darel13712/rs_datasets/issues/9

- Your environment?
`datatable == 1.0.0, python==3.7.7, Ubuntu 18.04.2`

I attach the files with the example lines below. They are named `csv` because github doesn't like `tsc` extension, but they are actually **tab** separated.
[not_lost.csv](https://github.com/h2oai/datatable/files/7556454/not_lost.csv)
[lost_col.csv](https://github.com/h2oai/datatable/files/7556455/lost_col.csv)

P.S. Maybe it's linked to #2680

**Comment by oleksiyskononenko** [comment]
When I load the `lost_col.csv` in the latest datatable I don't have anything lost

```python
>>> dt.fread("lost_col.csv")
   | C0                                        C1                                    C2                      C3
   | str32                                     str32                                 str32                int32
-- + ----------------------------------------  ------------------------------------  -------------------  -----
 0 | 0142ae97f6a6cb78b68b10cf93e91558474b1dbb  7746d775-9550-4360-b8d5-c37bd448ce01  "weird al" yankovic    327
[1 row x 4 columns]
```

Could you please check the latest version and see if it resolves the problem? My feeling is that this bug could already be fixed some time ago.

**Comment by Darel13712** [comment]
Yes, I am positive that 'lost_col.csv' loads only 3 columns with `datatable==1.0.0`. I just checked it with a fresh venv. Also you can see it being reproduced on [colab](https://colab.research.google.com/drive/1zKLq8iUgEudjK7UB2oZwYdyrEV66Q-cD?usp=sharing).

Maybe the fix is on master, or dev version, but I can't reproduce this behaviour with the version from pypi.

**Comment by oleksiyskononenko** [comment]
Exactly. There is a bug in `1.0.0`, but datatable in `main` doesn't have it. So if you want to get rid of the bug right away, you need to install the most recent version. Either from sources, or from the latest wheel on [s3](https://h2o-release.s3.amazonaws.com/datatable/index.html).

**Comment by Darel13712** [comment]
Does the release candidate also solve the `Too few fields, expected 4, but found only 3` with the whole dataset?

**Comment by oleksiyskononenko** [comment]
I don't know, when I click the link to the whole dataset — nothing happens in my browser. If you already have the data, you can install the latest dt and test. We had some related issues in the past, there is a chance the issue has already been fixed.

**Comment by Darel13712** [comment]
Yeah. That's strange. Well, you can right click, copy link address, create new tab and paste that link there, then it prompts to download it.

But yes, this is solved on `datatable==1.1.0a2102`, so I guess I should close this now.

Any forecasts on when the new version will be released?
